functions/index.js

```diff
@@ -0,0 +1,58 @@
+// we need ES5 requires becauses this folder does not have babel configured
+
+const admin = require('firebase-admin')
+const functions = require('firebase-functions')
+
+exports.onDiagnoseCreationAddTimestamp = functions.firestore
+  .document('diagnoses/{diagnoseId}')
+  .onCreate((snapshot, context) => {
+    const timestamp = admin.firestore.FieldValue.serverTimestamp()
+    return snapshot.ref.update({
+      createdAt: timestamp,
+      updatedAt: timestamp
+    })
+  })
+
+exports.onDiagnoseUpdateModifyTimestamp = functions.firestore
```

Comment:
What does this function do?

Reply:
I left it for future modifications on Diagnoses (such as submitting answers inside a diagnose).